File: java/src/main/java/it/giannibombelli/wsc2026/payment/integration/RefundRequestIntegrationCommand.java
package it.giannibombelli.wsc2026.payment.integration;

import it.giannibombelli.wsc2026.common.domain.primitive.Money;
import it.giannibombelli.wsc2026.common.utils.Require;

/**
 * Published Language di {@code payment} per richiedere un rimborso: il {@code clientReference}
 * identifica il pagamento, senza esporre il modello interno del BC.
 */
public record RefundRequestIntegrationCommand(String clientReference, Money amount) {
    public RefundRequestIntegrationCommand {
        Require.requireArgument(clientReference, "clientReference");
        Require.requireArgument(amount, "amount");
    }
}

File: typescript/src/payment/integration/refundRequestIntegrationCommand.ts
// Published Language esposta dal BC payment per richiedere un rimborso.
// Usa solo tipi stabili (string, Money) in modo da essere invocabile
// da qualsiasi bounded context senza introdurre coupling sui tipi interni di payment.

import type { Money } from '@/common/domain/primitive/money.js';
import { requireArgument } from '@/common/utils/requireArgument.js';

export type RefundRequestIntegrationCommand = {
  readonly kind: 'RefundRequestIntegrationCommand';
  readonly clientReference: string;
  readonly amount: Money;
};

export function refundRequestIntegrationCommand(
  clientReference: string,
  amount: Money
): RefundRequestIntegrationCommand {
  requireArgument(clientReference, 'clientReference');
  requireArgument(amount, 'amount');
  return { kind: 'RefundRequestIntegrationCommand', clientReference, amount };
}
